pyproject.toml

```diff
@@ -4,14 +4,35 @@ version = "0.1.0"
 description = ""
 authors = ["Benjamin Cretois <benjamin.cretois@nina.no>"]
 readme = "README.md"
+package-mode = false
 
 [tool.poetry.dependencies]
-python = "^3.10"
-torch = "^2.3.1"
+python = "~=3.10.0"
+torch = "2.3.1"
 ultralytics = "^8.2.42"
 matplotlib = "^3.9.0"
 opencv-python = "^4.10.0.84"
+einops = "0.8.0"
+numpy = "2.0.0"
+PyYAML = "6.0.1"
+tensorboard = "2.17.0"
+tensorboard-data-server = "0.7.2"
+torchmetrics = "1.4.0.post0"
+torchvision = "0.18.1"
+tqdm = "4.66.4"
+gradio = "4.44.1"
+optuna = "^4.0.0"
 
+# Prebulit wheel https://github.com/facebookresearch/detectron2/discussions/5200
+detectron2 = {version = "0.6+2a420edpt2.3.1cu121", source = "torch_packages_builder"}
+
+# The following line should not be necessary; drop it by the end of October 2024
+#flash-attn = {url = "https://github.com/Dao-AILab/flash-attention/releases/download/v2.6.3/flash_attn-2.6.3+cu123torch2.3cxx11abiFALSE-cp312-cp312-linux_x86_64.whl"}
```

Comment:
You can drop these lines, since you have tested the functionality of the whole system.